backend/core/src/shared/units-transformations.ts

```diff
@@ -115,58 +111,79 @@ const hmiData = (units: Units, byUnitType: UnitSummary[], amiPercentages: string
+}
+
+const getRoundedNumber = (initialValue: string) => {
+  return parseFloat(parseFloat(initialValue).toFixed(2))
+}
+
+const getDefaultSummaryRanges = (unit: Unit) => {
+  return {
+    areaRange: { min: parseFloat(unit.sqFeet), max: parseFloat(unit.sqFeet) },
+    minIncomeRange: {
+      min: getCurrencyString(unit.monthlyIncomeMin),
+      max: getCurrencyString(unit.monthlyIncomeMin),
     },
-    parseFloat(unit.monthlyIncomeMin)
-  )
-  summary.minIncomeRange = {
-    min: minIncomeRange.min.toFixed(2),
-    max: minIncomeRange.max.toFixed(2),
+    occupancyRange: { min: unit.minOccupancy, max: unit.maxOccupancy },
+    rentRange: {
+      min: getCurrencyString(unit.monthlyRent),
+      max: getCurrencyString(unit.monthlyRent),
+    },
+    rentAsPercentIncomeRange: {
+      min: parseFloat(unit.monthlyRentAsPercentOfIncome),
+      max: parseFloat(unit.monthlyRentAsPercentOfIncome),
+    },
+    floorRange: {
+      min: unit.floor,
+      max: unit.floor,
+    },
+    unitType: unit.unitType,
+    totalAvailable: 0,
+  } as UnitSummary
+}
+
+const getUnitsSummary = (unit: Unit, existingSummary?: UnitSummary) => {
+  if (!existingSummary) {
+    return getDefaultSummaryRanges(unit)
   }
+  const summary = existingSummary
+
+  // Income Range
+  summary.minIncomeRange = minMaxCurrency(
+    summary.minIncomeRange,
+    getRoundedNumber(unit.monthlyIncomeMin)
+  )
+
+  // Occupancy Range
+  summary.occupancyRange = minMax(summary.occupancyRange, unit.minOccupancy)
+  summary.occupancyRange = minMax(summary.occupancyRange, unit.maxOccupancy)
 
-  summary.occupancyRange = minMaxValue(summary.occupancyRange, unit.minOccupancy, unit.maxOccupancy)
-  summary.rentAsPercentIncomeRange = minMaxValue(
+  // Rent Ranges
+  summary.rentAsPercentIncomeRange = minMax(
     summary.rentAsPercentIncomeRange,
     parseFloat(unit.monthlyRentAsPercentOfIncome)
   )
-  const rentRange = minMaxValue(
-    {
-      min: parseFloat(summary.rentRange?.min),
-      max: parseFloat(summary.rentRange?.max),
-    },
-    Number.parseFloat(unit.monthlyRent)
-  )
-  summary.rentRange = {
-    min: rentRange.min.toFixed(2),
-    max: rentRange.max.toFixed(2),
-  }
+  summary.rentRange = minMaxCurrency(summary.rentRange, getRoundedNumber(unit.monthlyRent))
+
+  // Floor Range
   if (unit.floor) {
-    summary.floorRange = minMaxValue(summary.floorRange, unit.floor)
+    summary.floorRange = minMax(summary.floorRange, unit.floor)
   }
-  summary.areaRange = minMaxValue(summary.areaRange, parseFloat(unit.sqFeet))
 
-  if (summary.minIncomeRange) {
-    summary.minIncomeRange = minMaxInCurrency(summary.minIncomeRange)
-  }
-  if (summary.rentRange) {
-    summary.rentRange = minMaxInCurrency(summary.rentRange)
-  }
+  // Area Range
+  summary.areaRange = minMax(summary.areaRange, parseFloat(unit.sqFeet))
+
   return summary
 }
 
 type UnitMap = {
   [key: string]: Unit[]
 }
 
+const UnitTypeSort = ["studio", "oneBdrm", "twoBdrm", "threeBdrm"]
```

Comment:
This should probably be moved to an enum that `unitType` also uses on the unit entity. I don't think we need to do this right now with this task though, so I created this issue https://github.com/bloom-housing/bloom/issues/1335